test/framework/options.py

```diff
@@ -2968,6 +2968,51 @@ def test_list_software(self):
         self.assertTrue(re.search('^\s+\* GCC v4.6.3: dummy', txt, re.M))
         self.assertFalse(re.search('gzip', txt, re.M))
 
+    def test_parse_optarch(self):
+        """Test correct parsing of optarch option."""
+        
+        # Check for EasyBuildErrors
+        args = ['--optarch=Intel:something;GCC']
+        error_msg = "The optarch option has an incorrect syntax"
+        self.assertErrorRegex(EasyBuildError, error_msg, self.eb_main, args, raise_error=True)
+
+        args = ['--optarch=Intel:something;GCC:somethingelse;Intel:anothersomething']
+        error_msg = "The optarch option contains duplicated entries for compiler"
+        self.assertErrorRegex(EasyBuildError, error_msg, self.eb_main, args, raise_error=True)
+
+        # Check that it doesn't raise an exception when the input is correct
+        test_easyconfig = 'easyconfigs/test_ecs/t/toy/toy-0.0.eb'
+        test_cases = [
+                ['--optarch=',test_easyconfig],
+                ['--optarch=something',test_easyconfig],
+                ['--optarch=GENERIC',test_easyconfig],
+                ['--optarch=Intel:something',test_easyconfig],
+                ['--optarch=Intel:something;GCC:somethingelse',test_easyconfig],
+                ['--optarch=Intel:GENERIC;GCC:somethingelse',test_easyconfig],
+                ['--optarch=Intel:;GCC:somethingelse;',test_easyconfig]
+                ]
+        for args in test_cases:
+            self.eb_main(args, raise_error=True)
```

Comment:
What you say is done in the tests of `toolchain.py`. There I check against `xFLAGS`.

What I wanted to check here is simply that there aren't any issues when the input is correct. Arguably, it doesn't make too much sense, since the affected code is tested right below, calling `options.postprocess()`, where the code is really validated.

The reason I included these apparently useless tests is because the tests below check against just the postprocessing of options, instead of a "full" run. If you think this is not necessary I can simply delete these lines, but a more thorough test is already done in `toolchain.py`, I see no point in repeating it.